src/models/workspace.ts

```diff
@@ -29,19 +29,15 @@ export const WorkspaceModel = types
   })
   .actions((self) => {
     const setPrimaryDocument = (document?: DocumentModelType) => {
+      self.primaryDocumentKey = document && document.key;
```

Comment:
@sfentress We learned something fun today! Deleting MST keys removes their listeners, which stopped React render updates from happening.

Reply:
Huh, very interesting...

So when you call `setPrimaryDocument` without a document, it now goes back to `self.primaryDocumentKey = false`. But this does not remove the listener?

Reply:
Right, the proxies that are attached to the property are set up to trigger on assignment to the property, but are eliminated on deletion of the property.

Reply:
Good to know, thanks for catching. :+1: